bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/accessories/AbstractHomekitAccessoryImpl.java

```diff
@@ -349,4 +354,58 @@ protected BooleanItemReader createBooleanReader(HomekitCharacteristicType charac
         return new BooleanItemReader(taggedItem.getItem(), taggedItem.isInverted() ? OnOffType.OFF : OnOffType.ON,
                 taggedItem.isInverted() ? OpenClosedType.CLOSED : OpenClosedType.OPEN);
     }
+
+    /**
+     * Calculates a string as json of the configuration for this accessory, suitable for seeing
+     * if the structure has changed, and building a dummy accessory for it.
+     */
+    public String toJson() {
+        var builder = Json.createArrayBuilder();
+        getServices().forEach(s -> {
+            builder.add(serviceToJson(s));
+        });
+        return builder.build().toString();
+    }
+
+    private JsonObjectBuilder serviceToJson(Service service) {
+        var serviceBuilder = Json.createObjectBuilder();
+        serviceBuilder.add("type", service.getType());
+        var characteristics = Json.createArrayBuilder();
+
+        service.getCharacteristics().stream().sorted((l, r) -> l.getClass().getName().compareTo(r.getClass().getName()))
+                .forEach(c -> {
+                    try {
+                        var cJson = c.toJson(0).get();
```

Comment:
toJson(0)
does it make iid in HAP json always 0? or do i miss something?

Reply:
understood. thank you for the explanation. if JSON is not used directly as HAP payload, then it should be fine.